test/sql/types/unsigned/test_unsigned_auto_cast.test

```diff
@@ -0,0 +1,103 @@
+# name: test/sql/types/unsigned/test_unsigned_auto_cast.test
```

Comment:
I would like to see more exhaustive tests of casts to VARCHAR and to DECIMAL. Particularly testing boundaries and overflows, e.g.

```sql
SELECT '255'::UTINYINT; -- ok
SELECT '256'::UTINYINT; -- error
SELECT (100::UTINYINT)::DECIMAL(3,0); -- ok
SELECT (100::UTINYINT)::DECIMAL(2,0); --error
```

As well as some tests that verify that decimal arithmetic with auto casts work as expected
```sql
SELECT 100::UTINYINT * 100::DECIMAL(3,0); -- 10000
```